go.mod

```diff
@@ -32,16 +29,102 @@ require (
 	github.com/opencontainers/image-spec v1.0.1
 	github.com/pkg/errors v0.9.1
 	github.com/sanathkr/go-yaml v0.0.0-20170819195128-ed9d249f429b
-	github.com/sergi/go-diff v1.1.0 // indirect
 	github.com/sirupsen/logrus v1.8.1
 	github.com/spf13/cobra v1.2.1
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.7.0
-	github.com/xeipuuv/gojsonpointer v0.0.0-20190905194746-02993c407bfb // indirect
 	golang.org/x/sync v0.0.0-20210220032951-036812b2e83c
 	gotest.tools v2.2.0+incompatible
 	gotest.tools/v3 v3.0.3
+)
+
+require (
+	github.com/Azure/go-ansiterm v0.0.0-20210617225240-d185dfc1b5a1 // indirect
+	github.com/Masterminds/semver v1.5.0 // indirect
+	github.com/Microsoft/go-winio v0.4.17 // indirect
```

Comment:
Do we know if there's any "overrides" in the list of `// indirect` dependencies? Overall I think the go1.17 format makes sense for "indirect dependencies that we didn't manage", but if there's any `// indirect` that was explicitly added to the list to force a "newer version than `go.mod` resolved automatically", it may be good to put those in the first list.

That way it's clear what versions are "managed" in this `go.mod` (first block), and what versions are set "automatically" (so can be "ignored" when reading the `go.mod`)

Also see https://github.com/moby/buildkit/pull/2331#discussion_r702737246

Reply:
I don't think we manually override any `// indirect` dependencies, I didn't even know you could do that, I tend to think of the `go.mod` file as a lockfile I shouldn't update manually but only use go tool (unless you want to change the go version... go modules...)

Reply:
> I don't think we manually override any // indirect dependencies, I didn't even know you could do that

Yes, so you can use it in situations where you know an indirect dependency should be on a newer version than what go.mod automatically finds. In that case, you can add a `// indirect` with the version you want it to be on. The `// indirect` will automatically disappear once any other dependency requires a newer (or same) version of that dependency.

I know we had a couple of those in buildkit, but if that's not the case here, then no worries

> It's really optimistic in my opinion to update the go.mod file like that, what will happen when go 1.18 or 19 or 20 comes and they decide they want to change the way they write this mod file, like what they did with the 1.17 version

I think even for Go 1.17, the split format is still _optional_ (you can give options to split or not), but time will tell of course (it should still be possible to leave comments after the `// indirect`, e.g.

```
github.com/foo/bar     // indirect // using a newer version to address #123454
```